src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/PostgresLedgerExtenderService.cs

```diff
@@ -1438,6 +1438,41 @@ void AggregateEntityResourceVaultInternal(long entityId, long resourceEntityId,
                 }
             }
 
+            var resourceOwnersToAdd = entityResourceAggregatedVaultsHistoryToAdd
```

Comment:
Actually the source collection is guaranteed to be ordered by `state_version` so we could simplify all of that to:
```
var resourceOwnersToAdd = new Dict(..);

foreach (var x in entityResourceAggregatedVaultsHistoryToAdd) {
  resourceOwnersToAdd.AddOrUpdate(
    (x.EntityId, x.ResourceEntityId), 
    _ => new NonFungibleResourceOwners(...), 
    existing => existing.Balance = ...)
}
```